ignite/base/events_driven.py

```diff
@@ -270,14 +272,32 @@ def __init__(
         attr_to_events: Optional[Dict[str, List["Events"]]] = None,
         **kwargs: Any,
     ):
-
-        self._attr_to_events = attr_to_events if attr_to_events else defaultdict(list)  # type: Dict[str, List[Events]]
+        copy_attr_to_events = copy(attr_to_events)
```

Comment:
Why can't you copy the dict with `dict(attr_to_events)` ?

Reply:
We must check if it's not None first, can't decide copying or checking is better, what do you think?
```py
if attr_to_events:
            attr_to_events = dict(attr_to_events)
self._attr_to_events = (
    attr_to_events if attr_to_events else defaultdict(list)
)  # type: Dict[str, List[Events]]
```

Reply:
You can do the in a single line